From 67ad7c6e9169b872c8ca9f0ef8e873c66154ccc0 Mon Sep 17 00:00:00 2001
From: Gatien Chevallier <gatien.chevallier@foss.st.com>
Date: Tue, 2 Jul 2024 11:09:53 +0200
Subject: [PATCH 0995/1695] plat-stm32mp2: replace CFG_WARN_INSECURE usage

Replace occurrences of CFG_WARN_INSECURE by CFG_INSECURE on stm32mp
platforms as CFG_WARN_INSECURE is deprecated.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 6ed607836452cab746196efdf39c1aab381171e6 plat-stm32mp2: replace CFG_WARN_INSECURE usage]
Change-Id: I0756cc5bf64e069c867f25f948f8ba56f99280b7
Signed-off-by: Gatien Chevallier <gatien.chevallier@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/390744
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
Reviewed-by: Etienne CARRIERE <etienne.carriere@foss.st.com>
Tested-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
---
 core/arch/arm/plat-stm32mp1/conf.mk | 2 +-
 core/arch/arm/plat-stm32mp2/conf.mk | 2 +-
 core/arch/arm/plat-stm32mp2/main.c  | 2 +-
 core/drivers/stm32mp_provisioning.c | 4 ++--
 4 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/core/arch/arm/plat-stm32mp1/conf.mk b/core/arch/arm/plat-stm32mp1/conf.mk
index 22b443d42..ca9d5db77 100644
--- a/core/arch/arm/plat-stm32mp1/conf.mk
+++ b/core/arch/arm/plat-stm32mp1/conf.mk
@@ -347,7 +347,7 @@ endif
 # Provisioning support for BSEC shadow OTP is dedicated to insecure
 # development configuration only.
 ifeq ($(CFG_STM32MP_PROVISIONING),y)
-$(call force,CFG_WARN_INSECURE,y,Required by CFG_STM32MP_PROVISIONING)
+$(call force,CFG_INSECURE,y,Required by CFG_STM32MP_PROVISIONING)
 endif
 
 #SAES and CRYP cannot be register at the same time in the crypto framework
diff --git a/core/arch/arm/plat-stm32mp2/conf.mk b/core/arch/arm/plat-stm32mp2/conf.mk
index 1e1554a92..cc4ba5b5f 100644
--- a/core/arch/arm/plat-stm32mp2/conf.mk
+++ b/core/arch/arm/plat-stm32mp2/conf.mk
@@ -267,7 +267,7 @@ CFG_PKCS11_TA ?= y
 # configuration only.
 CFG_STM32MP_PROVISIONING ?= y
 ifeq ($(CFG_STM32MP_PROVISIONING),y)
-$(call force,CFG_WARN_INSECURE,y,Required by CFG_STM32MP_PROVISIONING)
+$(call force,CFG_INSECURE,y,Required by CFG_STM32MP_PROVISIONING)
 endif
 
 # Optional behavior upon receiving illegal access events
diff --git a/core/arch/arm/plat-stm32mp2/main.c b/core/arch/arm/plat-stm32mp2/main.c
index 2831aa3a1..a222ab234 100644
--- a/core/arch/arm/plat-stm32mp2/main.c
+++ b/core/arch/arm/plat-stm32mp2/main.c
@@ -182,7 +182,7 @@ static TEE_Result init_debug(void)
 		return res;
 
 	if (state != BSEC_STATE_SEC_CLOSED) {
-		if (IS_ENABLED(CFG_WARN_INSECURE))
+		if (IS_ENABLED(CFG_INSECURE))
 			IMSG("WARNING: All debug access are allowed");
 
 		res = stm32_bsec_write_debug_conf(BSEC3_DEBUG_ALL);
diff --git a/core/drivers/stm32mp_provisioning.c b/core/drivers/stm32mp_provisioning.c
index 35ef96237..9fc2cddf3 100644
--- a/core/drivers/stm32mp_provisioning.c
+++ b/core/drivers/stm32mp_provisioning.c
@@ -6,7 +6,7 @@
  * to provision BSEC/BSEC3 shadow cells with data read from OP-TEE OS
  * DTB but without any insurance that the OP-TEE driver are initialized
  * after this provisioning sequence. Therefore the driver initialization
- * prints a warning trace or panics upon CFG_WARN_INSECURE value.
+ * prints a warning trace or panics upon CFG_INSECURE value.
  */
 
 #include <arm.h>
@@ -216,7 +216,7 @@ static TEE_Result provisioning_probe(void)
 	provisioning_init();
 
 	if (!SLIST_EMPTY(&shadow_otp_head)) {
-		if (IS_ENABLED(CFG_WARN_INSECURE))
+		if (IS_ENABLED(CFG_INSECURE))
 			IMSG("WARNING: Embeds insecure stm32mp_provisioning driver");
 		else
 			panic("Embeds insecure stm32mp_provisioning data");
-- 
2.47.2

